ProcessLib/CoupledSolutionsForStaggeredScheme.cpp

```diff
@@ -18,48 +18,46 @@
 namespace ProcessLib
 {
 CoupledSolutionsForStaggeredScheme::CoupledSolutionsForStaggeredScheme(
-    std::unordered_map<std::type_index, Process const&> const&
-        coupled_processes_,
-    std::unordered_map<std::type_index, GlobalVector const&> const& coupled_xs_,
-    const double dt_)
-    : coupled_processes(coupled_processes_), coupled_xs(coupled_xs_), dt(dt_)
+    std::vector<std::reference_wrapper<GlobalVector const>> const& coupled_xs_,
+    const double dt_, const int process_id_)
+    : coupled_xs(coupled_xs_), dt(dt_), process_id(process_id_)
 {
-    for (auto const& coupled_x_pair : coupled_xs)
+    for (auto const& coupled_x : coupled_xs)
     {
-        auto const& coupled_x = coupled_x_pair.second;
-        MathLib::LinAlg::setLocalAccessibleVector(coupled_x);
+        MathLib::LinAlg::setLocalAccessibleVector(coupled_x.get());
     }
+}
+
+std::vector<std::vector<double>> getPreviousLocalSolutions(
+    const CoupledSolutionsForStaggeredScheme& cpl_xs,
+    const std::vector<GlobalIndexType>& indices)
+{
+    const auto number_of_coupled_solutions = cpl_xs.coupled_xs.size();
+    std::vector<std::vector<double>> local_xs_t0;
+    local_xs_t0.reserve(number_of_coupled_solutions);
 
-    for (auto const& coupled_process_pair : coupled_processes)
+    for (std::size_t i = 0; i < number_of_coupled_solutions; i++)
     {
-        auto const& coupled_pcs = coupled_process_pair.second;
-        auto const prevous_time_x = coupled_pcs.getPreviousTimeStepSolution();
-        if (prevous_time_x)
-        {
-            MathLib::LinAlg::setLocalAccessibleVector(*prevous_time_x);
-        }
+        auto const& x_t0 = *cpl_xs.coupled_xs_t0[i];
```

Comment:
for-range-loop again; I'm not sure about the type of '(*cpl_xs.coupled_xs_t0)' but if it has `begin` and `end`, then:
```c++
for (auto const& x_t0 : *cpl_xs.coupled_xs_t0)
{
   local_xs_t0...
}

Reply:
Changed accordingly.